Clean/Table/Basic.lean

```diff
@@ -125,249 +125,249 @@ end TraceOfLength
   To make sure that the vertical offset is bounded, it is represented as a `Fin W`.
 -/
 structure CellOffset (W: ℕ+) (S : Type → Type) [ProvableType S]  where
-  rowOffset: Fin W
+  row: Fin W
   column: Fin (size S)
-deriving Repr
+
+instance : Repr (CellOffset W S) where
+  reprPrec off _ := "⟨" ++ reprStr off.row ++ ", " ++ reprStr off.column ++ "⟩"
 
 namespace CellOffset
 
 /--
   Current row offset
 -/
-@[table_norm]
+@[table_assignment_norm]
 def curr {W : ℕ+} (j : Fin (size S)) :  CellOffset W S := ⟨0, j⟩
 
 /--
   Next row offset
 -/
-@[table_norm]
+@[table_assignment_norm]
 def next {W : ℕ+} (j : Fin (size S)) :  CellOffset W S := ⟨1, j⟩
 
 end CellOffset
 
-/--
-  Mapping from the index of a variable to a cell offset in the table.
--/
-@[reducible]
-def CellAssignment (W: ℕ+) (S : Type → Type) [ProvableType S] := ℕ → CellOffset W S
-
-/--
-  Atomic operations for constructing a table constraint, which is a constraint applied to a window
-  of rows in a table.
--/
-inductive TableConstraintOperation (W : ℕ+) (S : Type → Type) (F : Type) [Field F] [ProvableType S] where
```

Comment:
we no longer have a separate type for table operations! a table operation is either a circuit operation, and/or it directly changes the `ctx.assignment`